src/api/ecdar_api.rs

```diff
@@ -1,33 +1,29 @@
-use std::sync::Arc;
-
-use crate::api::server::server::get_auth_token_request::user_credentials;
-use crate::entities::access;
-use crate::entities::session;
 use chrono::Local;
 use regex::Regex;
 use sea_orm::SqlErr;
+use std::sync::Arc;
 use tonic::{Code, Request, Response, Status};
 
-use crate::api::server::server::{ecdar_api_auth_server::EcdarApiAuth, ecdar_api_server::EcdarApi};
-use crate::database::access_context::AccessContextTrait;
-use crate::database::in_use_context::InUseContextTrait;
-use crate::database::model_context::ModelContextTrait;
-use crate::database::query_context::QueryContextTrait;
-use crate::database::session_context::SessionContextTrait;
-use crate::database::user_context::UserContextTrait;
-use crate::entities::query;
-use crate::entities::user::Model as UserEntity;
-
-use super::server::server::get_auth_token_request::UserCredentials;
-use super::{
-    auth,
-    server::server::{
-        ecdar_backend_server::EcdarBackend, CreateAccessRequest, CreateQueryRequest,
-        CreateUserRequest, DeleteAccessRequest, DeleteQueryRequest, GetAuthTokenRequest,
-        GetAuthTokenResponse, QueryRequest, QueryResponse, SimulationStartRequest,
-        SimulationStepRequest, SimulationStepResponse, UpdateAccessRequest, UpdateQueryRequest,
-        UpdateUserRequest, UserTokenResponse,
-    },
+use crate::api::auth::{RequestExt, Token, TokenType};
+use crate::database::{
+    access_context::AccessContextTrait, in_use_context::InUseContextTrait,
+    model_context::ModelContextTrait, query_context::QueryContextTrait,
+    session_context::SessionContextTrait, user_context::UserContextTrait,
+};
+
+use super::server::server::{
+    ecdar_api_auth_server::EcdarApiAuth,
+    ecdar_api_server::EcdarApi,
+    ecdar_backend_server::EcdarBackend,
+    get_auth_token_request::{user_credentials, UserCredentials},
+    CreateAccessRequest, CreateQueryRequest, CreateUserRequest, DeleteAccessRequest,
+    DeleteQueryRequest, GetAuthTokenRequest, GetAuthTokenResponse, QueryRequest, QueryResponse,
+    SimulationStartRequest, SimulationStepRequest, SimulationStepResponse, UpdateAccessRequest,
+    UpdateQueryRequest, UpdateUserRequest, UserTokenResponse,
+};
+use crate::entities::{
+    access::Model as AccessEntity, query::Model as QueryEntity, session::Model as SessionEntity,
```

Comment:
Please use `x::Model` directly in the code instead of `XEntity` as there already is a ´x::entity´ which is used in the context files.

``` suggestion
access, query, session, user
```